python/pyspark/sql/dataframe.py

```diff
@@ -2046,6 +2046,40 @@ def toDF(self, *cols):
         jdf = self._jdf.toDF(self._jseq(cols))
         return DataFrame(jdf, self.sql_ctx)
 
+    @since(3.0)
+    def transform(self, func):
+        """Returns a new class:`DataFrame` according to a user-defined custom transform method.
+        This allows chaining transformations rather than using nested or temporary variables.
+
+        :param func: a user-defined custom transform function
+        This is equiavalent to a nested call:
+            actual_df = with_something(with_greeting(source_df), "crazy"))
```

Comment:
I wouldn't add this in the doc since doctests already describe the example.

Reply:
isn't doctest part of the doc?